src/metabase/util/password.clj

```diff
@@ -47,17 +50,41 @@
                  (when (>= (occurances char-type) min-count)
                    (recur more)))))))
 
-(def active-password-complexity
+(defn active-password-complexity
   "The currently configured description of the password complexity rules being enforced"
+  []
   (merge (complexity->char-type->min (config/config-kw :mb-password-complexity))
          ;; Setting MB_PASSWORD_LENGTH overrides the default :total for a given password complexity class
          (when-let [min-len (config/config-int :mb-password-length)]
            {:total min-len})))
 
-(def ^{:arglists '([password])} is-complex?
+(defn- is-complex?
   "Check if a given password meets complexity standards for the application."
-  (partial password-has-char-counts? active-password-complexity))
+  [password]
+  (password-has-char-counts? (active-password-complexity) password))
+
+(def ^File common-passwords-file
+  "A set of ~12k common passwords to reject, that otherwise meet Metabase's default complexity requirements.
+   Sourced from Dropbox's zxcvbn repo: https://github.com/dropbox/zxcvbn/blob/master/data/passwords.txt"
+  (io/file (io/resource "common_passwords.txt")))
+
+(defn- is-uncommon?
+  "Check if a given password is not present in the common passwords set. Case-insensitive search since
+   the list only contains lower-case passwords."
+  [password]
+  (with-open [reader (java.io.BufferedReader. (java.io.FileReader. common-passwords-file))]
```

Comment:
If you `:import` classes in the `ns` declaration you don't need to use the fully-qualified name. You can just do `(FileReader. ...)` for example. 

Also, you don't need to explicitly `:import` Java classes in Clojure if you use their fully-qualified name. The Clojure compiler will implicitly import them when you use them. The only reason to use `:import` is so you don't have to use the fully-qualified class name in the code in that namespace. Right now the `:import` statements here aren't doing anything